packages/fxa-auth-server/lib/routes/support.js

```diff
@@ -0,0 +1,96 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+'use strict';
+
+const error = require('../error');
+const isA = require('joi');
+const ScopeSet = require('fxa-shared').oauth.scopes;
+
+module.exports = (log, db, config, customs, zendeskClient) => {
+  // Skip routes if the subscriptions feature is not configured & enabled
+  if (!config.subscriptions || !config.subscriptions.enabled) {
+    return [];
+  }
+
+  const SUBSCRIPTIONS_MANAGEMENT_SCOPE =
+    'https://identity.mozilla.com/account/subscriptions';
+
+  async function handleAuth(auth, fetchEmail = false) {
+    const scope = ScopeSet.fromArray(auth.credentials.scope);
+    if (!scope.contains(SUBSCRIPTIONS_MANAGEMENT_SCOPE)) {
+      throw error.invalidScopes('Invalid authentication scope in token');
+    }
+    const { user: uid } = auth.credentials;
+    let email;
+    if (!fetchEmail) {
+      ({ email } = auth.credentials);
+    } else {
+      const account = await db.account(uid);
+      ({ email } = account.primaryEmail);
+    }
+    return { uid, email };
+  }
+
+  return [
+    {
+      method: 'POST',
+      path: '/support/ticket',
+      options: {
+        auth: {
+          payload: false,
+          strategy: 'oauthToken',
+        },
+        validate: {
+          payload: isA.object().keys({
+            topic: isA.string().required(),
+            subject: isA.string().optional(),
+            message: isA.string().required(),
+          }),
+        },
+        response: {
+          schema: isA.object().keys({
+            success: isA.bool().required(),
+            ticket: isA.number().optional(),
+            error: isA.string().optional(),
+          }),
+        },
+      },
+      handler: async function(request) {
+        log.begin('support.ticket', request);
+        const { uid, email } = await handleAuth(request.auth, true);
```

Comment:
There doesn't appear to be an instance where `fetchEmail` is false in handleAuth (maybe it will in future?). If not, you could update this to always fetch the account.

Reply:
I copied this method from the `subscriptions.js`, ideally it would be refactored out somewhere that both of these could use it. The current code architecture isn't very amenable to this though.

Reply:
Could it be pushed into the auth strategy? Or if not, just imported from some common helper module?